keps/sig-node/4753-expose-ownerref-downard-api/README.md

```diff
@@ -0,0 +1,616 @@
+	"metadata.name",
+	"metadata.namespace",
+	"metadata.labels",
+	"metadata.annotations",
+	"metadata.uid",
+  "metadata.ownerReferences")
+```
+
+At `pkg/apis/core/pods/helpers.go` we will edit the following function:
+
+```go
+func ConvertDownwardAPIFieldLabel(version, label, value string) (string, string, error) {
+	if version != "v1" {
+		return "", "", fmt.Errorf("unsupported pod version: %s", version)
+	}
+
+	if path, _, ok := fieldpath.SplitMaybeSubscriptedPath(label); ok {
+		switch path {
+		case "metadata.annotations", "metadata.labels":
+			return label, value, nil
+		default:
+			return "", "", fmt.Errorf("field label does not support subscript: %s", label)
+		}
+	}
+
+	switch label {
+	case "metadata.annotations",
+		"metadata.labels",
+		"metadata.name",
+		"metadata.namespace",
+		"metadata.uid",
+    "metadata.ownerReferences",
+		"spec.nodeName",
+		"spec.restartPolicy",
+		"spec.serviceAccountName",
+		"spec.schedulerName",
+		"status.phase",
+		"status.hostIP",
+		"status.hostIPs",
+		"status.podIP",
+		"status.podIPs":
+		return label, value, nil
+	// This is for backwards compatibility with old v1 clients which send spec.host
+	case "spec.host":
+		return "spec.nodeName", value, nil
+	default:
+		return "", "", fmt.Errorf("field label not supported: %s", label)
+	}
+}
+```
+* Information will be passed to the pod via the `valueFrom` field in the volume.
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: kubernetes-downwardapi-volume-example
+  ownerReferences:
+  - apiVersion: apps/v1
+    blockOwnerDeletion: true
+    controller: true
+    kind: DaemonSet
+    name: an-owned-pod-1722852739
+    uid: 6be1683f-da9c-4f68-9440-82376231cfa6
+  - apiVersion: apps/v1
+    blockOwnerDeletion: true
+    controller: true
+    kind: Some-CRD
+    name: an-owned-pod-1722852739
+    uid: 6be1683f-da9c-4f68-9440-82376231cfa6
+spec:
+  containers:
+    - name: client-container
+      image: busybox
+      command: ["sh", "-c"]
+      args:
+      - while true; do
+          if [[ -e /etc/podinfo/ownerReferences ]]; then
+            echo -en '\n\n'; cat /etc/podinfo/ownerReferences; fi;
+          sleep 5;
+        done;
+      volumeMounts:
+        - name: podinfo
+          mountPath: /etc/podinfo
+  volumes:
+    - name: podinfo
+      downwardAPI:
+        items:
+          - path: "ownerReferences"
+            fieldRef:
+              fieldPath: metadata.ownerReferences
+```
+
+Logs from the pod will show the ownerReferences of the pod:
+
+```bash
+$ kubectl logs kubernetes-downwardapi-volume-example
+- apiVersion: apps/v1
+  blockOwnerDeletion: true
+  controller: true
+  kind: DaemonSet
+  name: an-owned-pod-1722852739
+  uid: 6be1683f-da9c-4f68-9440-82376231cfa6
+- apiVersion: apps/v1
+  blockOwnerDeletion: true
+  controller: true
+  kind: Some-CRD
+  name: an-owned-pod-1722852739
+  uid: 6be1683f-da9c-4f68-9440-82376231cfa6
+```
+
+If `ownerReferences` is empty or not set, the file will be empty.
```

Comment:
what happen if `ownerReferences` is updated? is it reflected in the Pod or the Pod keeps using the original value?

Reply:
If this were to happen, isn't the Daemonset responsibility to refresh all the owned pods? and during this update cycle the injected file would be updated.

Reply:
I know very little about sig apps side of it. But is there any mutable fields on replicaset or DaemonSet that would not change the pod spec? Change of those will not be reflected in the downward API object.

Whenever we have any mutable value we lock in time, we like to add something in status to indicate which version of that value we have written down.

Maybe I misunderstood @aojea 's question.

Reply:
Also behavior on Container restart would be interesting. Whether we lock the value in time of Pod creation or we re-populate on every Container restart